presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -344,14 +344,14 @@ public void transferMemoryToTaskContext(long taskBytes)
         LocalMemoryContext taskLocalMemoryContext = driverContext.getPipelineContext().getTaskContext().localUserMemoryContext();
         try {
             operatorMemoryContext.localUserMemoryContext().transferOwnership(taskLocalMemoryContext);
-            taskLocalMemoryContext.addBytes(taskBytes - bytes);
+            taskLocalMemoryContext.setBytes(taskLocalMemoryContext.getBytes() + taskBytes - bytes);
         }
         catch (ExceededMemoryLimitException e) {
             // Since we can reserve an extra amount of memory above (when taskBytes > bytes) ExceededMemoryLimitException can be thrown
             // when reserving this extra amount. This will happen after "bytes" is moved from the operator context to the task context.
             // So, at this point "bytes" won't be tracked by the operator context and when the driver is destroyed it won't
             // be freed up from the memory pools.
-            taskLocalMemoryContext.addBytes(-bytes);
+            taskLocalMemoryContext.setBytes(taskLocalMemoryContext.getBytes() - bytes);
```

Comment:
For this case, I acknowledge that you actually want delta value． Let's talk about this.